src/main/java/com/yannbriancon/interceptor/HibernateQueryInterceptor.java

```diff
@@ -70,16 +68,29 @@ public void startQueryCount() {
         threadQueryCount.set(0L);
     }
 
+    /**
+     * Start or reset the query count to 0 for the considered thread.
+     * Counts only if given filter matches.
+     *
+     * @param filter string filter predicate
+     */
+    public void startQueryCount(Predicate<String> filter) {
+        this.filter = filter;
+        startQueryCount();
+    }
+
     /**
      * Get the query count for the considered thread
      */
     public Long getQueryCount() {
+        this.filter = null;
```

Comment:
There is no reason to reset the filter on getQueryCount.
Calling several times getQueryCount without having called startQueryCount between is broken.

Reply:
fixed